examples/components/MNIST/traininsilo/run.py

```diff
@@ -77,6 +92,48 @@ def __init__(
             self.test_dataset_, batch_size=batch_size, shuffle=True
         )
 
+        # DP
+        logger.info(f"DP: {dp}")
+        if dp:
+            if not ModuleValidator.is_valid(self.model_):
+                self.model_ = ModuleValidator.fix(self.model_)
+
+        self.optimizer_ = SGD(self.model_.parameters(), lr=lr, momentum=0.9)
+
+        if dp:
+            privacy_engine = PrivacyEngine(secure_mode=False)
+            """secure_mode: Set to True if cryptographically strong DP guarantee is
+            required. secure_mode=True uses secure random number generator for
+            noise and shuffling (as opposed to pseudo-rng in vanilla PyTorch) and
+            prevents certain floating-point arithmetic-based attacks.
+            See :meth:~opacus.optimizers.optimizer._generate_noise for details.
+            When set to True requires torchcsprng to be installed"""
+            (
+                self.model_,
+                self.optimizer_,
+                self.train_loader_,
+            ) = privacy_engine.make_private_with_epsilon(
+                module=self.model_,
+                optimizer=self.optimizer_,
+                data_loader=self.train_loader_,
+                epochs=dp_epochs,
+                target_epsilon=dp_target_epsilon,
+                target_delta=dp_target_delta,
```

Comment:
Minor suggestion: it might be helpful to log/print out the corresponding "noise multiplier" for such private_engine with the targeted epsilon and delta. You could get that "noise multiplier" value by optimizer.noise_multiplier after above definition of private_engine. It also helps explain a bit that epsilon, delta and noise multiplier are not independent from each other (we could compute the 3rd one given the other two).

Reply:
agree! done